gc.c

```diff
@@ -5901,13 +5924,16 @@ gc_compact_start(rb_objspace_t *objspace)
     struct heap_page *page = NULL;
 
     for (int i = 0; i < SIZE_POOL_COUNT; i++) {
-        rb_heap_t *heap = SIZE_POOL_EDEN_HEAP(&size_pools[i]);
+        rb_size_pool_t *pool = &size_pools[i];
+        rb_heap_t *heap = SIZE_POOL_EDEN_HEAP(pool);
         list_for_each(&heap->pages, page, page_node) {
             page->flags.before_sweep = TRUE;
         }
 
         heap->compact_cursor = list_tail(&heap->pages, struct heap_page, page_node);
         heap->compact_cursor_index = 0;
+
+       pool->compacted = 0;
```

Comment:
Nit: indentation looks off, and it would be nicer to use `FALSE` instead of the value `0`.